packages/terra-menu/src/MenuItem.jsx

```diff
@@ -54,36 +70,50 @@ const propTypes = {
   onClick: PropTypes.func,
 
   /**
-   * Callback function for when selection state changes on a selectable item.
+   * Callback function for when toggleable state changes on a toggleabe item.
    */
   onChange: PropTypes.func,
 
   /**
    * Indicates if the item has focus. This is used internally to control focus and does not set initial focus.
    */
   isActive: PropTypes.bool,
+
+  /**
+   * Custom icon to display in the item
+   */
+  icon: PropTypes.element,
 };
 
 const defaultProps = {
   text: '',
-  isSelected: false,
+  isToggled: false,
   isInstructionsForUse: false,
   isActive: false,
-  isSelectable: undefined,
+  isToggleable: undefined,
   isDisabled: false,
   subMenuItems: [],
 };
 
 class MenuItem extends React.Component {
   constructor(props, context) {
     super(props, context);
+    const {
+      isToggled,
+      isToggleable,
+      ...customProps
+    } = this.props;
+
     this.wrapOnClick = this.wrapOnClick.bind(this);
     this.wrapOnKeyDown = this.wrapOnKeyDown.bind(this);
     this.wrapOnKeyUp = this.wrapOnKeyUp.bind(this);
-    this.handleSelection = this.handleSelection.bind(this);
+    this.handleToggled = this.handleToggled.bind(this);
     this.setItemNode = this.setItemNode.bind(this);
+    const toggled = (isToggled || customProps.isSelected);
+    const toggleable = (isToggleable || customProps.isSelectable);
```

Comment:
```suggestion
    const toggled = (isToggled || customProps.isSelected);
    const toggleable = (isToggleable || customProps.isSelectable);
```
Lets move this variables outside of constructor so that it would accessed by other methods in this class.

Reply:
I cannot move these variables outside the constructor, in menuItemGroup, these items are cloned and the props can change, right now its just the toggled prop that is changing.  Therefore, I need to always get it fresh in the render method and is the safest way.
https://github.com/cerner/terra-framework/blob/main/packages/terra-menu/src/MenuItemGroup.jsx#L60